clusterloader2/pkg/measurement/common/resource_usage.go

```diff
@@ -104,34 +103,48 @@ func (e *resourceUsageMetricMeasurement) Execute(config *measurement.Measurement
 		}
 
 		klog.Infof("%s: starting resource usage collecting...", e)
-		e.gatherer, err = gatherers.NewResourceUsageGatherer(config.ClusterFramework.GetClientSets().GetClient(), host, provider, gatherers.ResourceGathererOptions{
-			InKubemark:                        strings.ToLower(provider) == "kubemark",
-			Nodes:                             nodesSet,
-			ResourceDataGatheringPeriod:       60 * time.Second,
-			MasterResourceDataGatheringPeriod: 10 * time.Second,
-			PrintVerboseLogs:                  false,
-		}, nil)
-		if err != nil {
-			return nil, err
+		for _, host := range hosts {
+			gatherer, err := gatherers.NewResourceUsageGatherer(config.ClusterFramework.GetClientSets().GetClient(), host, provider, gatherers.ResourceGathererOptions{
```

Comment:
I checked the code of the ContainerResourceGatherer and it's a bit messy (sorry about that!) It looks like `host` field is used only when the `InKubemark` option is true. Otherwise the host field is ignored and depending on the `Nodes` strategy different set of nodes is being monitored in terms of resource usage.

Given that, could you first refactor the `container_resource_gatherer.go` file to make the `host` field optional (part of the `ResourceGathererOptions`). Actually we can name it KubemarkMasterHost (and remove `InKubemark` field) to clean it up a bit. Once we do it it should be easier to refactor this file. 

Thanks!

Reply:
Absolutely, will start working on that! and thank you for your review.

Reply:
Thanks @alejandrox1 !